src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,201 @@
+
+    private static TaskList tasks = new TaskList();
+
+    public static void main(String[] args) throws IOException {
+        ui.printGreeting();
+        readAndExecuteCommand();
+    }
+
+    /**
+     * Reads and executes command from user inputs.
+     */
+    private static void readAndExecuteCommand() throws IOException {
+        Scanner in = new Scanner(System.in);
+        tasks.tasks = storage.readDataFromFile();
+        while (true) {
+            String userInput = getUserInput(in);
+            try {
+                executeCommand(userInput);
+            } catch (DukeException e) {
+                exceptionHandler.handleDukeExceptions(e);
+            } catch (Exception e) {
+                exceptionHandler.handleOtherExceptions(e);
+            }
+        }
+    }
+
+    /**
+     * Reads the text entered by the user.
+     *
+     * @param in The Scanner to read user input.
+     * @return line Full line entered by the user.
+     */
+    private static String getUserInput(Scanner in) {
+        String line = in.nextLine();
+        return line;
+    }
+
+    /**
+     * Executes the command as specified by the user input.
+     *
+     * @param userInput Raw input from user.
+     */
+    private static void executeCommand(String userInput) throws DukeException, IOException {
+        final String[] commandTypeAndParams = parser.splitCommandAndArgs(userInput);
+        final String commandType = commandTypeAndParams[0];
+        final String commandArgs = commandTypeAndParams[1];
+
+        switch (commandType) {
+        case COMMAND_ADD_TODO:
+            executeAddTodo(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_ADD_DEADLINE:
+            executeAddDeadline(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_ADD_EVENT:
+            executeAddEvent(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_LIST:
+            executeList();
+            break;
+        case COMMAND_MARK_DONE:
+            executeMarkDone(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_DELETE:
+            executeDelete(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_EXIT:
+            executeExit();
+            break;
+        default:
+            throw new DukeException();
+        }
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    /**
+     * Executes exiting program.
+     */
+    private static void executeExit() throws IOException {
+        ui.printGoodBye();
+        System.exit(0);
+    }
+
+    /**
+     * Executes marking task as done.
+     *
+     * @param commandArgs Processed input arguments from user.
+     */
+    private static void executeMarkDone(String commandArgs) {
+        int taskIndex = Integer.parseInt(commandArgs.strip()) - 1;
```

Comment:
Good and clear naming of variables! Likewise for most of the other variables I see in your functions